pilot/pkg/networking/core/v1alpha3/listenermatch.go

```diff
@@ -0,0 +1,86 @@
+package v1alpha3
+
+import (
+	xdsapi "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+
+	networking "istio.io/api/networking/v1alpha3"
+)
+
+func findMatches(listenerMatch *networking.EnvoyFilter_ListenerMatch, builder *ListenerBuilder) []*xdsapi.Listener {
+	var matches []*xdsapi.Listener
+	listeners := builder.getListeners()
+
+LISTENER_MATCH:
+	for _, listener := range listeners {
+		if listenerMatch.GetName() != "" && listenerMatch.GetName() != listener.GetName() {
+			continue
+		}
+
+		socketAddress := listener.Address.GetSocketAddress()
+		if (socketAddress == nil || socketAddress.GetPortValue() > 0) &&
+			listenerMatch.GetPortNumber() != socketAddress.GetPortValue() {
+			continue
+		}
+
+		if (socketAddress == nil || socketAddress.GetNamedPort() != "") &&
+			listenerMatch.GetPortName() != socketAddress.GetNamedPort() {
```

Comment:
this is incorrect. The named port thing is not implemented in Envoy. Its for a different purpose altogether, and has no bearing on the port names used by Istio. Lets remove this.